packages/react-reconciler/src/ReactFiberSuspenseComponent.js

```diff
@@ -23,3 +24,17 @@ export type SuspenseState = {|
   // `didTimeout` because it's not set unless the boundary actually commits.
   timedOutAt: ExpirationTime,
 |};
+
+export function shouldCaptureSuspense(
```

Comment:
I split this out because it was getting hard to follow all the nested branches in the `throwException` function